c10/core/WrapDimMinimal.h

```diff
@@ -6,7 +6,9 @@ namespace c10 {
 
 static inline int64_t maybe_wrap_dim(int64_t dim, int64_t dim_post_expr, bool wrap_scalar=true) {
   if (dim_post_expr <= 0) {
-    AT_CHECK(wrap_scalar, "dimension specified as ", dim, " but tensor has no dimensions");
+    if (!wrap_scalar) {
+        AT_INDEX_ERROR("dimension specified as ", dim, " but tensor has no dimensions");
```

Comment:
this changes this case of operating over an invalid dimension to be an AT_INDEX_ERROR, but the case below isn't changed.

Reply:
Thanks, done in #17114!

 I think there are more cases in `src/ATen/WrapDimUtils.h; unfortunately these affect `test_sparse` and it is not always clear to me whether `ValueError` or `IndexError` should be raised.  In `test_indexing` the choice was straightforward except for one case where NumPy raises `ValueError` (I marked that one in a comment).